core/controllers/suggestion.py

```diff
@@ -284,28 +284,32 @@ def _require_valid_suggestion_and_target_types(
             raise self.InvalidInputException(
                 'Invalid suggestion_type: %s' % suggestion_type)
 
-    def _render_suggestions(self, target_type, suggestions):
+    def _render_suggestions(self, target_type, suggestions, next_offset):
         """Renders retrieved suggestions.
 
         Args:
             target_type: str. The suggestion type.
             suggestions: list(BaseSuggestion). A list of suggestions to render.
+            next_offset: int. The number of results to skip in a subsequent
```

Comment:
I wonder if you could make it clearer here (and possibly in other relevant docstrings) that this is the number of results skipped from the beginning of the entire backend list of all suggestions matching the query parameters, and has no correlation to the current fetch? When I first read it, I thought it was the number of results to skip in the next fetch because they had been pre-retrieved in the current fetch (e.g. let's say I make a fetch for 20 results, the backend returns 25, so the offset for the next fetch is 5 -- we sometimes do some of this "over-returning" for complex queries, such as random-generation of questions for a practice questions session where we have to split the query into multiple subqueries).

(I developed a better understanding while subsequently reading the rest of this PR but if I was looking at this file in isolation I think it might have taken me a while to figure it out.)